### ISSUE TITLE
MeshAgent not reconnecting after UAC

### ISSUE THREAD

**Issue by dxdemetriou** [Ylianst/MeshAgent#150]
When is not installed but only connected, if you try to run something asking for UAC (for example cmd run as admin) and accepted by the local user, the agent is not reconnecting back (the screen seems freezed) and the supporter needs to disconnect and reconnect every time after a UAC prompt.

The agent itself tried with high privileges (by accepting UAC prompt), and tried also without accepting UAC. Without accepting the UAC is not crashing now as older versions, just cannot interact with high privileged apps. The UAC prompt is not visible, but I know that it's necessary to be run as service for this.

I'll try to make more tests tomorrow, it's better to try the master version here?

**Comment by krayon007** [comment]
The issue here, is that when the UAC pops up, the screen capture fails and exits. Normally what the agent does, is reconnect as local system. However, of course if the agent is running as a local user, it won't be able to do this, which is why it's failing, because when it tries to switch back, it specifies the SID of the local user, but that API only works with local system. 

What I can do, is try to make it so it doesn't try to switch users. That should make it so that when somebody accepts the UAC, the screen capture resumes.

I can start to take a look at that next week, to see if that is feasible

**Comment by dxdemetriou** [comment]
It seems that partially work is already done.

With MeshAgent & UAC:
-Open cmd with run-as admin
-Screen freezes but you can interact with the desktop
-Disconnect/Reconnect on MeshCentral and everything is visible again

==> It seems that only the screen freezes but you can interact.
==> Reconnecting from MeshCentral fixes the freeze.

With MeshAgent & No UAC
-Open cmd with run-as admin
-Screen freezes and you cannot interact with anything on desktop
-Disconnect/Reconnect on MeshCentral, everything is visible but you cannot interact with anything until the UAC app is closed
-Disconnect/Reconnect the MeshAgent and after connecting you can interact with the desktop except UAC apps 

==> It seems that the screen freezes and you cannot interact.
==> Reconnecting from MeshCentral fixes the freeze but not the interaction until either the UAC app is closed or the MesgAgent disconnects and reconnects.

I tried both master & current versions, it's almost the same behaviour

**Comment by dxdemetriou** [comment]
"UAC apps" I mean the application after the local used pressed the "Yes", not the UAC window. The UAC window I know that is vissible only after installation.
